src/app/schedule/schedule.component.ts

```diff
@@ -19,15 +21,33 @@ export class ScheduleComponent{
   arrivalPoint: string;
   price: number;
 
-  trips: Trip[] = [];
+  trips: ITrip[] = LocalStorageService.getItem('trips') ? LocalStorageService.getItem('trips') : [];
 
   addTrip(): void {
     this.trips = [{
       departureTime: this.departureTime,
       departurePoint: this.departurePoint,
       arrivalPoint: this.arrivalPoint,
-      price: this.price
-    }, ...this.trips];
+      price: this.price,
+      isEditable: false
+    } as ITrip, ...this.trips];
+    LocalStorageService.setItem('trips', this.trips);
   }
-}
+  deleteTrip(trip: ITrip): void {
+    this.trips = this.trips.filter(obj => obj !== trip);
+    LocalStorageService.setItem('trips', this.trips);
+  }
+  editTrip(trip: ITrip): void {
+    let valid = true;
+    document.querySelector('#tripTable').querySelectorAll('input').forEach(el => {
```

Comment:
use form object instead or validate directly variables assigned to corresponding fields